src/weights.jl

```diff
@@ -475,13 +559,22 @@ function wsum(A::AbstractArray{T}, w::AbstractVector{W}, dim::Int) where {T<:Num
     _wsum!(similar(A, wsumtype(T,W), Base.reduced_indices(axes(A), dim)), A, w, dim, true)
 end
 
+function wsum(A::AbstractArray{<:Number}, w::UnitWeights{S,T}, dim::Int) where {S,T}
```

Comment:
No need for parameters:
```suggestion
function wsum(A::AbstractArray{<:Number}, w::UnitWeights, dim::Int)
```

Same elsewhere.